controller/src/controller.rs

```diff
@@ -918,7 +920,7 @@ impl Controller {
 
     /// Report the status of a set of transceiver modules.
     pub async fn status(&self, modules: ModuleId) -> Result<StatusResult, Error> {
```

Comment:
Thinking out loud, it might help the transition for this to attempt the `StatusV2` message, falling back to `Status` if that fails in a particular way. I think we can tell if a message is unsupported, though we've had few chances to test that.

Reply:
I decided to allow for a more graceful transition by leaving `Controller::status` as-is, and adding `Controller::extended_status` to get the wider `ExtendedStatus` message.  Dendrite can start using the latter in a later patch, instead of breaking now.